docs/README.md

```diff
@@ -1,3 +1,201 @@
+
+All properties bound in C# access the value through accessor functions. If a Swift property or a Swift method mutates the value, the contents of the C# `SwiftData` property will get changed as well.
+
+### Enums
+
+Swift enums are projected into C# as `IDisposable` classes, implementing `ISwiftEnum`, in similar manner as structs.
+
+Given this Swift enum:
+```swift
+    public enum FooECTIA {
+        case a(Int)
+        case b(Double)
+    }
+```
+
+The projection tooling will generate the following C#, with function bodies left empty for simplicity:
+```csharp
+    using System;
+    using System.Runtime.InteropServices;
+    using SwiftRuntimeLibrary;
+    using SwiftRuntimeLibrary.SwiftMarshal;
+    
+    namespace NewClassCompilerTests
+    {
+        public enum FooECTIACases
+        {
+            A, B
+        }
+        [SwiftEnumType("libNewClassCompilerTests.dylib",
+            "$s21NewClassCompilerTests8FooECTIAOMn", 
+            "$s21NewClassCompilerTests8FooECTIAON", "")]
+        public class FooECTIA : ISwiftEnum
+        {
+            public void Dispose()
+            {
+            }
+            void Dispose(bool disposing)
+            {
+            }
+            ~FooECTIA()
+            {
+            }
+            public static FooECTIA NewA(nint value0)
+            {
+            }
+            public static FooECTIA NewB(double value0)
+            {
+            }
+            public byte[] SwiftData
+            {
+                get; set;
+            }
+            public nint ValueA
+            {
+                get { }
+            }
+            public double ValueB
+            {
+                get { }
+            }
+            public FooECTIACases Case
+            {
+                get { }
+            }
+        }
+    }
+```
+
+### Scaling and trimming
+
+The bindings should be organized into namespaces to mitigate scalability issues. Additionally, generated bindings should be trim-compatible.
+
+This subsection will be expanded during the design review process.
+
+### Unsupported types
+
+This subsection describes cases when direct interop is not possible and will be expanded during the design review process. This is a subset of features that require Swift wrappers provided by users:
+- Protocols and classes with virtual methods
+- Closures
+- Passing a struct by value in more registers than P/Invoke will allow
+- Exporting .NET into Swift
+
+## Functional outline
+
+The tooling consists of the following components:
+- **CLIInterface**: A command-line interface that orchestrates the tooling workflow.
+- **SwiftReflector**: A component that aggregates public API definitions from Swift modules.
+- **Dynamo**: A component that provides API for C# code generation.
```

Comment:
Any particular reason Roslyn wouldn't work?

Reply:
FWIW, Roslyn is not a particularly great API for generating source code. Most source generators generate the source by printing it one line at time, without much help from Roslyn. Random example: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs#L461-L485

Reply:
Another facet of this that has been mentioned is that source generators aren't the best solution when C# isn't the "source of truth". When the C# is the "source of truth", the source generators are appropriate - see `LibraryImport` or the COM source generator.

Reply:
Added to the list in https://github.com/dotnet/runtime/issues/95633.